src/o365/graph/commands/teams/teams-channel-set.ts

```diff
@@ -0,0 +1,149 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelName: string;
+  newChannelName: string;
+  description: string
+}
+class GraphTeamsChannelSetCommand extends GraphItemsListCommand<Channel>{
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_SET}`;
+  }
+  public get description(): string {
+    return 'Updates properties of a specified channel in the given Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.role = args.options.teamId;
```

Comment:
Hi @waldekmastykarz I always wanted to ask why we use this function. I think I did not catch the purpose.

Reply:
We track the usage how the different commands in the CLI are being used. For required options, there is no need to track them, because they're always present, but we are interested in which optional arguments are used.